src/java.desktop/share/classes/java/awt/Graphics2D.java

```diff
@@ -1200,10 +1200,14 @@ public abstract boolean hit(Rectangle rect,
      * previously been set, or if the clip has been cleared using
      * {@link Graphics#setClip(Shape) setClip} with a {@code null}
      * argument, the specified {@code Shape} becomes the new
-     * user clip.
+     * user clip. Since this method intersects the specified shape
+     * with the current clip, it will throw {@code NullPointerException}
+     * for a null shape unless the user clip is also {@code null}.
+     * So calling this method with a null argument is not recommended.
```

Comment:
Since you're editing the javadoc for this method, wouldn't adding a couple `<p>` break up the description to make it clearer?

```
1198     * <p>The <i>user clip</i> modified by this method is independent of the
1203     * user clip.
         * <p>Since this method intersects the specified shape
```
The rendered javadoc will have clear separation between different paragraphs and will facilitate scanning the method description.

Does it make any sense?

Reply:
ok. sure...